process/smartContract/process.go

```diff
@@ -1250,12 +1238,9 @@ func (sc *scProcessor) createSCRsWhenError(
 		if callType == vmcommon.AsynchronousCall {
 			scr.CallType = vmcommon.AsynchronousCallBack
 			scr.GasPrice = tx.GetGasPrice()
-			sc.mutGasLock.RLock()
-			gasToLock := sc.asyncCallStepCost + sc.asyncCallbackGasLock
-			sc.mutGasLock.RUnlock()
-			if tx.GetGasLimit() >= gasToLock {
-				scr.GasLimit = gasToLock
-				consumedFee = core.SafeMul(tx.GetGasPrice(), tx.GetGasLimit()-gasToLock)
+			if tx.GetGasLimit() >= gasLocked {
+				scr.GasLimit = gasLocked
+				consumedFee = core.SafeMul(tx.GetGasPrice(), tx.GetGasLimit()-gasLocked)
```

Comment:
If dynamic gas locking is enabled, then `tx.GetGasLimit()` is all the gas that has to be consumed, because the locked gas is kept separately. No need to subtract it here, this can be just `tx.GetGasLimit()`.

Reply:
this is correct. only the remaining was consumed in this shard.